src/Psalm/Internal/Codebase/Populator.php

```diff
@@ -374,74 +466,73 @@ private function populateOverriddenMethods(
         }
     }
 
-    private function populateDataFromTraits(
+    private function populateDataFromTrait(
```

Comment:
I pulled the loops out of these functions because in a previous version I was running them individually on specific dependencies. I later realized that line 295 already reruns the entire thing anyway, but I left the changes here because there were other improvements done as well.